### PR TITLE
xds: Added IncompleteData interface in mock CEL library

### PR DESCRIPTION
**Pull request by cindyxue**
This PR added `IncompleteData` interface into the mock CEL library in gRPC-Java.

CEL `eval` function will return an `IncompleteData` object if there are missing attributes.

@ZhenLian

### CODE REVIEW

**Comment by ZhenLian** [comment]
@sanjaypujare Hi Sanjay, can you please take a look at this PR please? We find one more CEL mock interface that we want to add. Thank you so much!

File: xds/src/main/java/io/grpc/xds/internal/rbac/engine/cel/IncompleteData.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.internal.rbac.engine.cel;
+
+/**
+ * Add an interface for interpreter to check if an object is an instance
```
Comment (sanjaypujare) [inline xds/src/main/java/io/grpc/xds/internal/rbac/engine/cel/IncompleteData.java:20]: "Add an ..." sounds like the PR description copied as Javadoc comment here. Shouldn't this be something like "Interpreter returns an instance of `{@code IncompleteData}` if an object is an instance of `{@link PartialMessage}`" . 

I am assuming `PartialMessage` is defined somewhere.

Comment (ZhenLian) [reply xds/src/main/java/io/grpc/xds/internal/rbac/engine/cel/IncompleteData.java:20]: Thanks for the review! `PartialMessage` is defined internally, and from API-wise we don't need it at this moment. Regarding to the comment, We copied them from the internal CEL lib, but I agreed that the original comments could be improved, and yours are much better... @cindyxue Can you please update them? Thanks!

Comment (cindyxue) [reply xds/src/main/java/io/grpc/xds/internal/rbac/engine/cel/IncompleteData.java]: Thank you both for the review and the comments! Fixed :)

**Pull request merged**